### PR TITLE
start copy and paste for midi notes

### PR DESCRIPTION
**Pull request by RobinMontferme**
Started to implement code to copy and paste Midi note in the Midi Piano Roll process from issue #727

### CODE REVIEW

File: src/plugins/score-plugin-midi/Midi/MidiNoteEditor.cpp [hunk 1]
```diff
@@ -1,22 +1,98 @@
...
+
+  // TODO this is a bit lazy.. find a better positoning algorithm7
+  //Position in the case the pasting is done through ui
+  //  if (!mv.contains(mv_pt))
+  //    mv_pt = mv.mapToScene(mv.boundingRect().center());
+
+  // Read the copy json. TODO: give it a better mime type
+  //  auto origin = pres->toScenarioPoint(sv_pt);
+  auto obj = readJson(mime.data("text/plain"));
+  JSONWriter w(obj);
```
Comment (jcelerier) [inline src/plugins/score-plugin-midi/Midi/MidiNoteEditor.cpp:75]: add an if(!obj.IsObject()) { return false; } here as this will crash otherwise

**Review by jcelerier — APPROVED** [review]
all good so far, just make sure to check the curve copy paste (CurveEditor.cpp) for some of the refactored parts

**Comment by jcelerier** [comment]
code looks good so far ! I think that for the TODO of keepign the notes selected, you can do it in a further PR ?

**Comment by RobinMontferme** [comment]
Sure absolutely !

**Pull request merged**